File: Chapter 05/JS/section_5_4_server_monitor/after/infrastructure/adapters/kafkaAdapter.js
const AlertPort = require('../../core/ports/alertPort');

/**
 * ADAPTER 3: The "Scale" Adapter (Async Messaging).
 * Shows how easy it is to swap a "Sync" SMS for an "Async" message.
 */
class KafkaAdapter extends AlertPort {
    constructor(kafkaProducer) {
        super();
        this.kafkaProducer = kafkaProducer;
    }

    sendAlert(message) {
        const payload = JSON.stringify({
            Error: message,
            Timestamp: new Date().toISOString()
        });
        
        // Fire and forget
        this.kafkaProducer.produce("server-alerts-topic", payload);
    }
}

module.exports = KafkaAdapter;

File: Chapter 05/Python/section_5_4_server_monitor/after/infrastructure/adapters/kafka_adapter.py
import json
from datetime import datetime
from core.ports.alert_port import AlertPort

class KafkaAdapter(AlertPort):
    """
    ADAPTER 3: The 'Scale' Adapter (Async Messaging).
    Shows how easy it is to swap a 'Sync' SMS for an 'Async' message.
    """

    def __init__(self, kafka_producer):
        self.kafka_producer = kafka_producer

    def send_alert(self, message: str) -> None:
        payload = json.dumps({
            "Error": message,
            "Timestamp": datetime.utcnow().isoformat()
        })
        
        # Fire and forget
        self.kafka_producer.produce("server-alerts-topic", payload)